subsys/bluetooth/host/classic/spp.c

```
@@ -0,0 +1,357 @@
+	}
+
+	if (response->resp_buf == NULL) {
+		LOG_ERR("SPP sdp resp_buf is null");
+		goto exit_err;
+	}
+
+	LOG_DBG("SPP SDP record found for uuid:%s", bt_uuid_str(params->uuid));
+
+	err = bt_sdp_get_proto_param(response->resp_buf, BT_SDP_PROTO_RFCOMM, &channel);
+	if (err < 0) {
+		LOG_ERR("SPP sdp get proto fail");
+		goto exit_err;
+	}
+
+	spp->acl_conn = bt_conn_ref(conn);
+	spp->rfcomm_dlc.ops = &spp_rfcomm_ops;
+	spp->rfcomm_dlc.mtu = SPP_RFCOMM_MTU;
+	spp->rfcomm_dlc.required_sec_level = BT_SECURITY_L2;
+
+	err = bt_rfcomm_dlc_connect(conn, &spp->rfcomm_dlc, channel);
+	if (err < 0) {
+		LOG_ERR("SPP rfconn connect fail, err:%d", err);
+		goto exit_err;
+	}
+
+	return BT_SDP_DISCOVER_UUID_STOP;
+
+exit_err:
+	if (spp->ops->disconnected) {
+		spp->ops->disconnected(spp);
+	}
+
+	return BT_SDP_DISCOVER_UUID_STOP;
+}
+
+int bt_spp_server_register(struct bt_spp_server *server)
+{
+	if (!server || (server->accept == NULL) || (server->sdp_record == NULL)) {
+		LOG_DBG("Invalid parameter");
+		return -EINVAL;
+	}
+
+	k_sem_take(&spp_lock, K_FOREVER);
+
+	if (sys_slist_find(&spp_rfcomm_server, &server->node, NULL)) {
+		k_sem_give(&spp_lock);
+		LOG_WRN("RFCOMM server has been registered");
+		return -EEXIST;
+	}
+
+	bt_sdp_register_service(server->sdp_record);
+
+	server->rfcomm_server.accept = bt_spp_accept;
+	bt_rfcomm_server_register(&server->rfcomm_server);
+
+	sys_slist_append(&spp_rfcomm_server, &server->node);
+
+	k_sem_give(&spp_lock);
+	return 0;
+}
+
+struct bt_sdp_record *bt_spp_alloc_record(uint8_t channel)
+{
+	static uint8_t index;
+	struct bt_sdp_record *sdp;
+
+	k_sem_take(&spp_lock, K_FOREVER);
+
+	if (index >= CONFIG_BT_SPP_MAX_SERVER_NUM) {
+		k_sem_give(&spp_lock);
+		LOG_ERR("No available SPP SDP record slots");
+		return NULL;
+	}
+
+	if ((channel < SPP_CHANNEL_MIN) || (channel > SPP_CHANNEL_MAX)) {
+		k_sem_give(&spp_lock);
+		LOG_ERR("Invalid RFCOMM channel: %d", channel);
+		return NULL;
+	}
+
+	spp_channel_map[index] = channel;
```

If there is a case that the same one channel has been allocated several times? Than the same channel will be registered. And it will cause the error when calling `bt_rfcomm_server_register()`. But the returned error is not checked in function `bt_spp_server_register()`.